lib/runtime/procs/read.js

```diff
@@ -222,6 +223,7 @@ var Read = source.extend({
             this.program.scheduler.schedule(nextRead.unixms(), _.bind(this.run, this));
         })
         .catch((err) => {
+            this.reading = null;
```

Comment:
We could -- one concern would be whether the scheduler could end up calling us back before the .finally handler executed. It'll probably work fine, but I'd rather keep things as-is with the duplication rather than rely on that timing to all line up properly.